src/scala/org/pantsbuild/zinc/compiler/OutputUtils.scala

```diff
@@ -0,0 +1,116 @@
+ */
+
+package org.pantsbuild.zinc.compiler
+
+import java.io.File
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{FileVisitResult, Files, Path, Paths, SimpleFileVisitor}
+import java.util.jar.{JarEntry, JarInputStream, JarOutputStream}
+import scala.annotation.tailrec
+import scala.collection.mutable
+import scala.util.Try
+
+object OutputUtils {
+
+  /**
+   * Sort the contents of the `dir` in lexicographic order.
+   *
+   * @param dir File handle containing the contents to sort
+   * @return sorted set of all paths within the `dir`
+   */
+  def sort(dir:File): mutable.TreeSet[Path] = {
+    val sorted = new mutable.TreeSet[Path]()
+
+    val fileSortVisitor = new SimpleFileVisitor[Path]() {
+      override def preVisitDirectory(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        sorted.add(path)
+        FileVisitResult.CONTINUE
+      }
+
+      override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        sorted.add(path)
+        FileVisitResult.CONTINUE
+      }
+    }
+
+    Files.walkFileTree(dir.toPath, fileSortVisitor)
+    sorted
+  }
+
+  def relativize(base: String, path: Path): String = {
+    new File(base).toURI().relativize(new File(path.toString).toURI()).getPath()
+  }
+
+  /**
+   * Create a JAR of of filePaths provided.
+   *
+   * @param filePaths set of all paths to be added to the JAR
+   * @param outputJarPath Absolute Path to the output JAR being created
+   * @param jarEntryTime time to be set for each JAR entry
+   */
+  def createJar(
+    base: String, filePaths: mutable.TreeSet[Path], outputJarPath: Path, jarEntryTime: Long) {
+
+    val target = new JarOutputStream(Files.newOutputStream(outputJarPath))
+
+    def addToJar(source: Path, entryName: String): FileVisitResult = {
+      val jarEntry = new JarEntry(entryName)
+      // setting jarEntry time to a fixed value for all entries within the jar so that jars are
+      // byte-for-byte reproducible.
+      jarEntry.setTime(jarEntryTime)
+
+      target.putNextEntry(jarEntry)
+      Files.copy(source, target)
+      target.closeEntry()
+      FileVisitResult.CONTINUE
+    }
+
+    val pathToName = filePaths.zipWithIndex.map{case(k, v) => (k, relativize(base, k))}.toMap
+    pathToName.map(e => addToJar(e._1, e._2))
+    target.close()
+  }
+
+  /**
+   * Jar the contents of output classes (settings.classesDirectory) and copy to settings.outputJar
+   *
+   */
+  def createClassesJar(classesDirectory: File, outputJarPath: Path, jarCreationTime: Long) = {
+
+    // Sort the contents of the classesDirectory for deterministic jar creation
+    val sortedClasses = sort(classesDirectory)
+
+    createJar(classesDirectory.toString, sortedClasses, outputJarPath, jarCreationTime)
+  }
+
+  /**
+   * Determines if a Class exists in a JAR provided.
+   *
+   * @param jarPath Absolute Path to the JAR being inspected
+   * @param clazz Name of the Class, the existence of which is to be inspected
```

Comment:
I think this function checks the existence of files in the jar and not the existence of classes by name. It might make sense to rename it and the `clazz` parameter to make that clearer.